sunpy/instr/goes.py

```diff
@@ -583,16 +612,489 @@ def _goes_get_chianti_em(longflux, temp, satellite=8, abundances="coronal",
+    # emission measure, calculate using calculate_temperature_em()
+    if 'temperature' in goeslc.data and 'em' in goeslc.data:
+        # Use copy.deepcopy for replicating meta and data so that input
+        # lightcurve is not altered.
+        lc_new = sunpy.lightcurve.LightCurve(meta=copy.deepcopy(goeslc.meta),
+                                             data=copy.deepcopy(goeslc.data))
+    else:
+        lc_new = calculate_temperature_em(goeslc)
+    temp = Quantity(np.asarray(lc_new.data.temperature, dtype=np.float64), unit=u.MK)
+    em = Quantity(np.asarray(lc_new.data.em, dtype=np.float64), unit=u.cm**(-3))
+
+    # Find radiative loss rate with _calc_rad_loss()
+    rad_loss_out = _calc_rad_loss(temp, em, force_download=force_download,
+                                  download_dir=download_dir)
+
+    # Enter results into new version of GOES LightCurve Object
+    lc_new.data["rad_loss_rate"] = rad_loss_out["rad_loss_rate"].to("W").value
+
+    return lc_new
+
+@quantity_input(temp=u.MK, em=u.cm**(-3))
+def _calc_rad_loss(temp, em, obstime=None, force_download=False,
+                   download_dir=DATA_PATH):
+    """
+    Finds radiative loss rate of coronal plasma over all wavelengths.
+
+    This function calculates the radiative loss rate of solar coronal
+    soft X-ray-emitting plasma across all wavelengths given an isothermal
+    temperature and emission measure.  The units of the results are
+    erg/s.  This function is based on calc_rad_loss.pro in SSW IDL.
+    In addition, if obstime keyword is set, giving the times to which
+    the temperature and emission measure values correspond, the
+    radiated losses integrated over time are also calculated.
+
+    Parameters
+    ----------
+    temp : `astropy.units.Quantity`
+        Array containing the temperature of the coronal plasma at
+        different times.  Units=[MK]
+
+    em : `astropy.units.Quantity`
+        Array containing the emission measure of the coronal plasma
+        at the same times corresponding to the temperatures in temp.
+        Must be same length as temp.  Units=[cm**-3]
+
+    obstime : (optional) array-like of datetime objects
+        Array of measurement times to which temperature and
+        emission measure values correspond.  Must be same length
+        as temp and em.  If this keyword is set, the integrated
+        radiated energy is calculated.
+
+    force_download : (optional) bool
+        If True, the GOES radiative loss data file is downloaded.  It is
+        important to do this if a new version of the files has been
+        generated due to a new CHIANTI version being released or the
+        launch of new GOES satellites.
+        Default=False
+
+    download_dir : (optional) string
+        The directory to download the GOES radiative loss data file to.
+        Default=SunPy default download directory
+
+    Returns
+    -------
+    radlossrate : numpy ndarray, dtype=float, units=[erg]
```

Comment:
This dosen't seem to be correct. It seems to be returning a dict of `Quantity` objects?